React/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -1,16 +1,45 @@
-export const actions = {
-  GET_LOGGED_USER: 'GET_LOGGED_USER',
-  SET_LOGGED_USER: 'SET_LOGGED_USER'
-};
+import { stringArrayToObject } from '../../utils/stringUtils';
+import { login as loginService } from '../../services/LoginService';
+
+export const actions = stringArrayToObject(
+  [
+    'LOGIN_REQUEST',
+    'LOGIN_SUCCESS',
+    'LOGIN_FAILURE'
+  ],
+  '@@LOGIN'
+);
 
 const actionCreators = {
-  getLoggedUser: () => ({
-    type: actions.GET_LOGGED_USER
-  }),
-  setLoggedUser: status => ({
-    type: actions.SET_LOGGED_USER,
-    logged: status
-  })
+  login: (user) => async dispatch => {
+    dispatch({ type: actions.LOGIN_REQUEST });
+    const response = await loginService({ email: user.email });
+    if (response.ok) {
+      if (response.data.length > 0) {
+        if (response.data[0].password === user.password) {
+          dispatch({
+            type: actions.LOGIN_SUCCESS,
+            payload: response.data
+          });
+        } else {
+          dispatch({
+            type: actions.LOGIN_FAILURE,
+            payload: 'Password is incorrect'
+          });
+        }
+      } else {
+        dispatch({
+          type: actions.LOGIN_FAILURE,
+          payload: 'Email not registered yet'
+        });
+      }
+    } else {
+      dispatch({
```

Comment:
throw an error.. use LOGIN_FAILURE just for the cases above

Reply:
mmmm let's leave it for now.. it will change in the recompose part

Reply:
great